frontend/server/src/Controllers/Tag.php

```diff
@@ -65,4 +65,34 @@ public static function getPublicTags() {
             'problemTag'
         );
     }
+
+    /**
+     * Return most frequent public tags of a certain level
+     *
+     */
+    public static function getFrequentTagsByLevel(
+        string $problemLevel
+    ): array {
+        return \OmegaUp\Cache::getFromCacheOrSet(
+            \OmegaUp\Cache::TAGS_LIST,
+            'frequent',
+            fn () => \OmegaUp\DAO\Tags::getFrequentsByLevel(
+                $problemLevel
+            ),
+            APC_USER_CACHE_SESSION_TIMEOUT
+        );
+    }
+
+    /**
+     * Return most frequent public tags of a certain level
+     *
+     * @return list<array{name: string}>
```

Comment:
Mismo caso que los otros comentarios:
```suggestion
     * @return list<array{alias: string}>
```